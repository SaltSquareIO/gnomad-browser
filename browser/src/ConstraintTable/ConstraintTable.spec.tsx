import React from 'react'
import { Factory } from 'fishery' // eslint-disable-line import/no-extraneous-dependencies

import renderer from 'react-test-renderer'
import { createRenderer as createShallowRenderer } from 'react-test-renderer/shallow'
import { forAllDatasets, forAllDatasetsExcept } from '../../../tests/__helpers__/datasets'

import ConstraintTable, { Gene, Transcript } from './ConstraintTable'

const geneFactory = Factory.define<Gene>(() => ({
  chrom: '13',
  transcripts: [],
}))

const transcriptFactory = Factory.define<Transcript>(() => ({
  transcript_id: 'dummy_transcript',
  transcript_version: '12.34.5',
  chrom: '13',
}))

forAllDatasets('ConstraintTable with "%s" dataset selected', (datasetId) => {
  describe('with a minimal gene', () => {
    test('has no unexpected changes', () => {
      const tree = renderer.create(
        <ConstraintTable datasetId={datasetId} geneOrTranscript={geneFactory.build()} />
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with a minimal transcript', () => {
    test('has no unexpected changes', () => {
      const tree = renderer.create(
        <ConstraintTable datasetId={datasetId} geneOrTranscript={transcriptFactory.build()} />
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with a mitochondrial gene', () => {
    test('has no unexpected changes', () => {
      const tree = renderer.create(
        <ConstraintTable
          datasetId={datasetId}
          geneOrTranscript={geneFactory.build({ chrom: 'M' })}
        />
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe('with a mitochondrial transcript', () => {
    test('has no unexpected changes', () => {
      const tree = renderer.create(
        <ConstraintTable
          datasetId={datasetId}
          geneOrTranscript={transcriptFactory.build({ chrom: 'M' })}
        />
      )
      expect(tree).toMatchSnapshot()
    })
  })
})

test('ConstraintTable with exac dataset and gene with available constraints has no unexpected changes', () => {
  const tree = createShallowRenderer().render(
    <ConstraintTable
      datasetId="exac"
      geneOrTranscript={geneFactory.build({ exac_constraint: {} })}
    />
  )
  expect(tree).toMatchSnapshot()
})

test('ConstraintTable with exac dataset and transcript with available constraints has no unexpected changes', () => {
  const tree = createShallowRenderer().render(
    <ConstraintTable
      datasetId="exac"
      geneOrTranscript={transcriptFactory.build({ exac_constraint: {} })}
    />
  )
  expect(tree).toMatchSnapshot()
})

forAllDatasetsExcept(['exac'], 'ConstraintTable with "%s" dataset selected', (datasetId) => {
  describe('and gene with available constraint', () => {
    test('has no unexpected changes', () => {
      const tree = createShallowRenderer().render(
        <ConstraintTable
          datasetId={datasetId}
          geneOrTranscript={geneFactory.build({ gnomad_constraint: {} })}
        />
      )
      expect(tree).toMatchSnapshot()
    })
  })

  describe('and transcript with available constraint', () => {
    test('has no unexpected changes', () => {
      const tree = createShallowRenderer().render(
        <ConstraintTable
          datasetId={datasetId}
          geneOrTranscript={transcriptFactory.build({ gnomad_constraint: {} })}
        />
      )
      expect(tree).toMatchSnapshot()
    })
  })
})
