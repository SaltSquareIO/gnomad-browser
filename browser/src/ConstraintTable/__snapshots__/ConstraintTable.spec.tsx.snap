// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConstraintTable with "exac" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "exac" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "exac" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "exac" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_controls" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_cancer" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_neuro" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_r2_1_non_topmed" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_controls_and_biobanks" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_cancer" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_neuro" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_topmed" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_r3_non_v2" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint not yet available for gnomAD v3.
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_controls" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected and gene with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected and transcript with available constraint has no unexpected changes 1`] = `
<React.Fragment>
  <p>
    Constraint is based on the full gnomAD dataset, not the selected subset.
  </p>
  <GnomadConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected with a minimal gene has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  gene
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected with a minimal transcript has no unexpected changes 1`] = `
<p>
  Constraint not available for this 
  transcript
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected with a mitochondrial gene has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  genes
</p>
`;

exports[`ConstraintTable with "gnomad_sv_r2_1_non_neuro" dataset selected with a mitochondrial transcript has no unexpected changes 1`] = `
<p>
  Constraint is not available for mitochondrial 
  transcripts
</p>
`;

exports[`ConstraintTable with exac dataset and gene with available constraints has no unexpected changes 1`] = `
<React.Fragment>
  <ExacConstraintTable
    constraint={Object {}}
  />
  <p>
    Constraint metrics based on 
    Ensembl canonical
     transcript (
    <withRouter()
      preserveSelectedDataset={true}
      to="/transcript/undefined"
    >
      .
    </withRouter()>
    ).
  </p>
</React.Fragment>
`;

exports[`ConstraintTable with exac dataset and transcript with available constraints has no unexpected changes 1`] = `
<React.Fragment>
  <ExacConstraintTable
    constraint={Object {}}
  />
</React.Fragment>
`;
